### PR TITLE
Us image posts

### PR DESCRIPTION
**Pull request by nodeJayS**
Controller method to upload images

### CODE REVIEW

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs [hunk 1]
```diff
@@ -233,5 +239,27 @@ public async Task<IActionResult> GetNewsfeedAsync()
             return NoContent();
         }
 
+        [HttpPost("UploadPicture"), DisableRequestSizeLimit]
+        public async Task<ActionResult> UploadPicture() 
+        {
+            try {
+                if (Request.Form.Files[0] is IFormFile file)
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:246]: just use model binding

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs [hunk 2]
```diff
@@ -233,5 +239,27 @@ public async Task<IActionResult> GetNewsfeedAsync()
...
+                {
+                    // generate a random guid from the file name
+                    var newFileName = $"{Request.Form["userId"]}-{Guid.NewGuid()}.{file.FileName.Split('.').Last()}";
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:249]: it'd be nice if an example file name were given to illustrate the format here. not a bad idea to factor that into a helper method either.

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs [hunk 3]
```diff
@@ -233,5 +239,27 @@ public async Task<IActionResult> GetNewsfeedAsync()
...
+                    // upload image
+                    var result = await _blobService.UploadFileBlobAsync(
+                        "fakebook",
+                        file.OpenReadStream(),
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:253]: we may be leaking unmanaged resources here, if this stream is something we need to dispose.

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs [hunk 4]
```diff
@@ -233,5 +239,27 @@ public async Task<IActionResult> GetNewsfeedAsync()
...
+                        file.ContentType,
+                        newFileName);
+                    return Ok(new { path = result.AbsoluteUri });
+                } else return BadRequest();
+            } catch (Exception ex) {
+                _logger.LogError(ex, ex.Message);
+                return BadRequest(ex.Message);
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:260]: arbitrary exceptions should not be caught and turned into bad request. only things that you can concretely attribute to the request should be bad request, the rest just don't catch them and let asp.net return 500. anyway some exceptions might have details in their message that shouldn't be told to the client for security.

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs (hunk 1 above)
Comment (nodeJayS) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs]: UploadPicture now takes a file and userId parameter

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs (hunk 2 above)
Comment (nodeJayS) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs]: Wrote a comment that shows what happens with a gif being uploaded

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs (hunk 3 above)
Comment (nodeJayS) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs]: Enclosed in a using statement now

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs (hunk 4 above)
Comment (nodeJayS) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:266]: Specified exceptions for invalid operation, request failed and a 500 for bad files

**Review by escalonn — APPROVED** [review]
missing authz, but if it works..

**Pull request merged**